src/routes/v2/collections.rs

```diff
@@ -231,7 +231,7 @@ pub async fn collection_edit(
     let result = database::models::Collection::get(id, &**pool, &redis).await?;
 
     if let Some(collection_item) = result {
-        if !is_authorized_collection(&collection_item, &user_option).await? {
+        if collection_item.user_id != user.id.into() && !user.role.is_mod() {
```

Comment:
Why aren't we using the common method any more? I don't think we want to be introducing more duplication if we can help it

Reply:
The issue here is that is_authorized_collection doesnt actually check if its a collection that *belongs* to you, only whether its visible to you. This was one of the bugs I found.

Reply:
Could we make a new method that does it then? Don't love that this line is copy/pasted 4 times across this file...

Reply:
Fair enough!